### PR TITLE
Tika client

### PR DESCRIPTION
**Pull request by SEQUOIIA**

### CODE REVIEW

File: src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                CapabilityId = capability.Id,
+                CapabilityName = capability.Name
+            };
+
+            await kafkaClient.SendMessageAsync(message, "capability_deleted");
+        }
+
+        private async Task Then_the_connected_acls_are_deleted(Capability capability, ServiceAccount serviceAccount)
+        {
+            var serviceAccountId = Convert.ToInt64(serviceAccount.Id);
```
Comment (kim-lindhard-dfds) [inline src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs]: This should live somewhere else

File: src/KafkaJanitor.IntegrationTests/TopicCreatedScenario.cs [hunk 2]
```diff
@@ -1,56 +1,56 @@
...
-                TopicName = "devex-integrationtest"
-            };
-            await _tikaRestClient.AddAsync(message.TopicName);
+                name =  "devex-integrationtest"
+            });
         }
 
         private async Task Then_a_topic_is_created()
         {
             Thread.Sleep(5 * 1000);
```
Comment (kim-lindhard-dfds) [inline src/KafkaJanitor.IntegrationTests/TopicCreatedScenario.cs]: Thead sleep smells

File: src/Infrastructure/KafkaJanitor.RestApi/Features/Topics/Infrastructure/TikaTopicRepository.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public async Task<bool> Exists(string topicName)
+        {
+            var topics = await _tikaClient.Topics.GetAllAsync();
+            try
+            {
+                topics.First(to => to.Equals(topicName));
+            }
+            catch (Exception)
```
[the remark this answers is not in the document]
Comment (SEQUOIIA) [reply src/Infrastructure/KafkaJanitor.RestApi/Features/Topics/Infrastructure/TikaTopicRepository.cs]: InvalidOperationException

File: src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+using Tika.RestClient.Features.Acls.Models;
+using Tika.RestClient.Features.ServiceAccounts.Models;
+using Xunit;
+using ServiceAccount = Tika.RestClient.Features.ServiceAccounts.Models.ServiceAccount;
+
+namespace KafkaJanitor.IntegrationTests
+{
+    public class CapabilityCreatedDeletedScenario
+    {
+        private IConfiguration _configuration;
```
[the remark this answers is not in the document]
Comment (SEQUOIIA) [reply src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs]: It is not. Aaaaaaaaaaaaand it's gone.

File: api-contract.yml [hunk 5]
```diff
@@ -1,57 +1,72 @@
...
-    properties:
-      name:
-        type: string
-        example: build.capabilities
-      partitions:
-        type: number
-        example: 12
-    
\ No newline at end of file
+swagger: '2.0'
```
[the remark this answers is not in the document]
Comment (SEQUOIIA) [reply api-contract.yml]: Updated it!

File: src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        {
+            serviceAccount = await _tikaClient.ServiceAccounts.CreateAsync(new ServiceAccountCreate {
+                name = $"{capability.Name}_sa",
+                description = "Creating during CapabilityCreatedDeletedScenario"
+                });
+        }
+
+        private async Task And_acls_to_create_read_write_to_topics_under_prefix(Capability capability, ServiceAccount serviceAccount)
+        {
+            var serviceAccountId = Convert.ToInt64(serviceAccount.Id);
```
[the remark this answers is not in the document]
Comment (SEQUOIIA) [reply src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs]: So it shall. https://app.clubhouse.io/devex/story/1371/change-kafka-janitor-integration-tests-to-use-services-startup-cs-for-tests-instead-of-the-clients-directly

File: src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs (hunk 1 above)
Comment (SEQUOIIA) [reply src/KafkaJanitor.IntegrationTests/CapabilityCreatedDeletedScenario.cs]: So it shall. https://app.clubhouse.io/devex/story/1371/change-kafka-janitor-integration-tests-to-use-services-startup-cs-for-tests-instead-of-the-clients-directly

**Comment by SEQUOIIA** [comment]
I think the K8S deployment manifest may be missing an environment variable for Tika. Let's take a look at it tomorrow.

**Pull request merged**